apps/www/_blog/2024-02-21-automating-performance-tests.mdx

```diff
@@ -0,0 +1,342 @@
+---
+title: 'Automating performance tests'
+description: Learn about our story to get to the automated performance testing.
+author: egor_romanov
+image: 2024-02-21-automating-performance-tests/automating-performance-tests-OG.png
+thumb: 2024-02-21-automating-performance-tests/automating-performance-tests-OG.png
+categories:
+  - engineering
+tags:
+  - performance
+date: '2024-02-21'
+toc_depth: 3
+---
+
+Performance testing is one of the non-functional testing types that evaluates a system's compliance with its performance requirements. It reveals if your app can handle user load, unexpected spikes, or recover from stressful workloads. In this blogpost you will learn about how we got to the automated performance testing. I hope our lessons will make this journey easier for you.
+
+At the end of this post you will have a practical approach for doing performance testing manually and automatically and a set of tools and templates to get started.
+
+## How this started?
+
+When I joined Supabase we had a challenging task of rewriting Realtime service so that users would be able to stream database updates with respect to RLS policies. Such a huge design change could not happen without having an impact on throughput.
+
+We decided to extensively test how the new Realtime would perform prior to launch to:
+
+- compare the capabilities of the new version and the current version,
+- determine the speed, performance, and stability of the new Realtime and establish standards for the future,
+- understand how it is scaling,
+- find bugs.
+
+In fact we had to test 2 versions at once for the new Realtime. The new architecture allowed it to work not only with a single database instance, but also as a multitenant independent service, distributed all over the world to bring presence and broadcast functionalities.
+
+<div>
+  <Img
+    alt="Realtime v1 & v2"
+    src={{
+      light: '/images/blog/2024-02-21-automating-performance-tests/realtime-v1-and-v2--light.png',
+      dark: '/images/blog/2024-02-21-automating-performance-tests/realtime-v1-and-v2.png',
+    }}
+  />
+</div>
+
+Realtime v1 & v2
+
+<div>
+  <Img
+    alt="Realtime v3"
+    src={{
+      light: '/images/blog/2024-02-21-automating-performance-tests/realtime-v3--light.png',
+      dark: '/images/blog/2024-02-21-automating-performance-tests/realtime-v3.png',
+    }}
+  />
+</div>
+
+Realtime v3
```

Comment:
It might be easier if the labels came on top of each images here.
```suggestion

Realtime v1 & v2:

<div>
  <Img
    alt="Realtime v1 & v2"
    src={{
      light: '/images/blog/2024-02-21-automating-performance-tests/realtime-v1-and-v2--light.png',
      dark: '/images/blog/2024-02-21-automating-performance-tests/realtime-v1-and-v2.png',
    }}
  />
</div>

Realtime v3:

<div>
  <Img
    alt="Realtime v3"
    src={{
      light: '/images/blog/2024-02-21-automating-performance-tests/realtime-v3--light.png',
      dark: '/images/blog/2024-02-21-automating-performance-tests/realtime-v3.png',
    }}
  />
</div>
```